src/IO/StdIStreamFromMemory.h

```diff
@@ -1,15 +1,15 @@
 #pragma once
 
-#include "config.h"
-
-#if USE_AWS_S3
-
 #include <iostream>
 
 namespace DB
 {
 
-struct MemoryStream: std::iostream
+/// StdIStreamFromMemory is used in WriteBufferFromS3 as a stream which is passed to the S3::Client
+/// It provides istream interface (only reading) over the memory.
+/// However S3::Client requires iostream interface it only reads from the stream
+
+class StdIStreamFromMemory : public std::iostream
```

Comment:
This class is a subclass of `std::iostream`, not `std::istream`.

Reply:
Yes, S3::Client is requires std::iostream interface. but writing to it is prohibited.

Reply:
Ok. Apparently it's bad design of AWS that they require `std::iostream` while they only need `std::istream`.